src/Microsoft.HttpRepl/FileSystem/RealFileSystem.cs

```diff
@@ -40,9 +42,31 @@ public void DeleteFile(string path)
             File.Delete(path);
         }
 
-        public string GetTempFileName()
+        public string GetTempFileName(string fileExtension = null)
         {
-            return Path.GetTempFileName();
+            // If they don't care about the extension, go the easy route and use the
+            // system-provided way of creating a unique temp file.
+            if (fileExtension is null)
+            {
+                return Path.GetTempFileName();
```

Comment:
There are now two fairly different behaviors of this method:

1. Called with no extension, it will create a 0-byte file on disk and return the path to it (what `Path.GetTempFileName()` does).  The new file is guaranteed to be unique as long as the system can provide one.
2. Called with an extension, it will return a string to a filename easily associated with http repl; however, no file is created and no check is made for duplicates (which is rare, but possible).

It feels like there's a design inconsistency here:
* should we always create the file?
* should we guarantee name uniqueness?
* should we always use the `HttpRepl.{Guid}` name, and use .tmp as a default extension?

Reply:
As far as the uniqueness goes, I was basing my "good enough" call on Jared's answer here on a similarly constructed temp file name: https://stackoverflow.com/a/581588/228298.
> Note, it's not guaranteed to produce a unique file name since someone else could have technically already created that file. However the chances of someone guessing the next guid produced by your app and creating it is very very low. It's pretty safe to assume this will be unique.

As far as whether the file gets created or not... I'd prefer it NOT get created... but there's no way to turn that off while still using `GetTempFileName()`, which brings me to the last point...

>should we always use the HttpRepl.{Guid}` name and use .tmp as a default extension?

Probably. I think that's the way I'm leaning. It gets rid of the unnecessarily created 0-byte file, and it keeps the consistency of being able to locate the file more easily.

If you're fine with the answer on uniqueness above, then I'll go that route.

Reply:
A uniqueness check could be done with a retry loop, but I agree that it's highly unlikely to happen, especially given the HttpRepl prefix.  Making the behavior consistent is the main change.